ffc-ahwr-mi-reporting/mi-report-v3/transformJsonToCsvV3.js

```diff
@@ -97,48 +98,65 @@ const multiHerdsColumns = [
   'herdReasonOther'
 ]
 
+const pigUpdatesColumns = [
+  'pigsElisaTestResult',
+  'pigsPcrTestResult',
+  'pigsGeneticSequencing'
+]
+
 const buildColumns = () => {
   return [
     ...defaultColumns,
-    ...(config.flagReporting.enabled ? flagColumns : []),
-    ...(config.multiHerds.enabled ? multiHerdsColumns : [])
+    ...flagColumns,
+    ...multiHerdsColumns,
+    ...(config.pigUpdates.enabled ? pigUpdatesColumns : [])
   ]
 }
 
-const getFlagData = (...args) => {
-  if (!config.flagReporting.enabled) {
+const getData = (...args) => {
+  return args
+}
+
+const getPigUpdatesData = (...args) => {
+  if (!config.pigUpdates.enabled) {
     return []
   }
 
   return args
 }
 
-const getHerdData = (...args) => {
-  if (!config.multiHerds.enabled) {
-    return []
+const formatPigsGeneticSequencing = async (geneticSequencingResult) => {
+  if (!geneticSequencingResult) {
+    return ''
   }
 
-  return args
+  const PIG_GENETIC_SEQUENCING_VALUES = await getPigGeneticSequencingValues()
+
+  const geneticSequencingLabel = PIG_GENETIC_SEQUENCING_VALUES.find(
+    (keyValuePair) => keyValuePair.value === geneticSequencingResult).label
+
+  return geneticSequencingLabel
 }
 
 // Function to transform event data to CSV row format
-function transformEventToCsvV3 (event, context) {
+async function transformEventToCsvV3 (event, context) {
```

Comment:
Do we need to make this async just to support getting those pig sequencing values? Wondering if we can perhaps instead have an async function to resolve the lookup values, before we kick off the transform, and then then format function that is currently async can be synchronous instead, using precomputed lookups, and then as a knock on this function can remain synchronous too? Would also mean we aren't resolving that for every row that is transformed too.

Probably doesn;t make much difference, but feels a big much to make the entire thing async to support getting those values resolved.

Reply:
I've taken the decision to just hardcode the values instead of this large workaround - it was ending up just not being worth it, as it affects basically all the test files.